### DISCUSSION TITLE
How to set relayPagination resolver for nested field?

### DISCUSSION THREAD

**Discussion by villesau** [urql-graphql/urql#2933] [Q&A] [answered]
Hi, 

Relay pagination is one of the most useful features of Urql. However, I've not found a way to use it in nested fields. Normally it works in the graph cache like this:

```
      resolvers: {
        Query: {
          someRelayList: relayPagination()
        }
    }
```

I'd expect it to work like this for nested fields:

```
      resolvers: {
        Query: {
          someObject: {
             someNestedRelayList: relayPagination()
          }
        }
    }
```

However, this gives a type error.

How to reap benefits of the relayPagination in the nested fields?

**Comment by kitten** [comment, marked as answer]
You can put the utility (and any corresponding resolver) on any field, provided you don't use `Query` but your sub-type's name, i.e. any other object types name,
so the config is basically set up to be `{ [typename: string]: { [fieldname: string]: resolverFn } }`

**Reply by villesau** [reply]
I tried to move to

```
      resolvers: {
        SomeObject: {
          someNestedRelayList: relayPagination()
        }
      },
```

but then the resolved `SomeObject` `data` is always null. Without the resolver I'm getting results normally, but the next page obviously fails.

The query looks like this:

```
query GetNestedList($someObjectId: ID!, $after: ID) {
  someObject(id: $someObjectId) {
    someNestedRelayList(after: $after) {
      pageInfo {
        hasNextPage
        endCursor
      }
      edges {
        cursor
        node {
          __typename
          ...AllFieldFragment
        }
      }
    }
  }
}
```

**Reply by villesau** [reply]
I think I figured out what's wrong. The relay paginator apparently expects `first` argument to be present, after passing that things started to work. Pothos uses default if it is not set: https://pothos-graphql.dev/docs/plugins/relay

So this worked:

```
query GetNestedList($someObjectId: ID!, $after: ID, $first: Int) {
  someObject(id: $someObjectId) {
    someNestedRelayList(after: $after, first: $first) {
      pageInfo {
        hasNextPage
        endCursor
      }
      edges {
        cursor
        node {
          __typename
          ...AllFieldFragment
        }
      }
    }
  }
}

```

Thanks for the help and fast response @kitten !

**Reply by villesau** [reply]
Filed a small PR to clarify the usage of relayPagination in the nested fields: https://github.com/urql-graphql/urql/pull/2934
